src/AnalyticsEventReporter.ts

```diff
@@ -34,10 +34,31 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
   with(payload: EventPayload): AnalyticsEventService {
     const currentPayload =
       this.payload === undefined ? payload : merge(this.payload, payload);
+    if (this.config.debug) {
+      console.log('[DEBUG] AnalyticsConfig Object:\n', this.config);
```

Comment:
I would move this up to the constructor since you dont have to call `with()` to make a request. When you move this to the constructor, I would also log the EventPayload as well

Reply:
I had this in the constructor at first, but then this ended up getting logged multiple times before I even clicked a button on the test-site. I think this is because `with` calls the constructor again. You are right that the constructor could be used without `with`, so I am going to move it back to the constructor, but do you think the below seems confusing or is it fine? 

![Screenshot 2024-02-14 at 5 06 05 PM](https://github.com/yext/analytics/assets/18221975/e179ff21-6740-4ee3-9056-481fced3b2d4)

Reply:
I would say that the logging should only show up when report is called.